tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -353,43 +325,42 @@ You can find more details about this in e.g. [5].
+Instead of drawing samples of each entry in $\mathbf{W}$ from a standard normal, this time we repeatedly draw $D$ samples from the $D$-dimensional MND, forming a $D \times D$ matrix $\mathbf{W}$.
 This matrix is a function of $\alpha$ as the samples are drawn from the MND parameterized by $\alpha$.
-We also introduce a hyper-parameter $tau$ which is the precision in the sampling distribution.
+We also introduce a hyper-parameter $\tau$ which is the precision in the sampling distribution.
+We also re-paramterise the sampling distribution, i.e. each dimension across all instances is a 60-dimensional multivariate normal distribution. Re-parameterisation can sometimes accelrate the sampling process.
 
-We instantiate the model and ask Turing to sample from it using NUTS sampler. The sample trajectories of $\alpha$ is plotted using the `plot` function from the packge `StatsPlots`.
+We instantiate the model and ask Turing to sample from it using NUTS sampler. The sample trajectories of $\alpha$ is plotted using the `plot` function from the package `StatsPlots`.
 
 ```julia
-ppca_ARD = pPCA_ARD(expression_matrix) # instantiate the probabilistic model
-chain_ppcaARD = sample(ppca_ARD, NUTS(), 500)
-
-StatsPlots.plot(group(chain_ppcaARD, :alpha))
+ppca_ARD = pPCA_ARD(mat_exp') # instantiate the probabilistic model
+chain_ppcaARD = sample(ppca_ARD, NUTS(), 500) # sampling
+StatsPlots.plot(group(chain_ppcaARD, :α))
 ```
 
-Again, we do some inference diagnostics. Here we look at the convergence of the chains for the $\alpha$ parameter. This parameter determines the relevance of individual components.
-We see that the chains have converged and the posterior of the alpha parameters is centered around much smaller values in two instances.
+Again, we do some inference diagnostics. Here we look at the convergence of the chains for the $α$ parameter. This parameter determines the relevance of individual components.
+We see that the chains have converged and the posterior of the $\alpha$ parameters is centered around much smaller values in two instances.
 In the following, we will use the mean of the small values to select the *relevant* dimensions (remember that, smaller values of $\alpha$ correspond to more important components.).
 We can clearly see from the values of $\alpha$ that there should be two dimensions (corresponding to $\bar{\alpha}_3=\bar{\alpha}_5≈0.05$) for this dataset.
 
 ```julia
 # Extract parameter mean estimates of the posterior
-w = permutedims(reshape(mean(group(chain_ppcaARD, :w))[:, 2], (n_genes, n_genes)))
-z = permutedims(reshape(mean(group(chain_ppcaARD, :z))[:, 2], (n_genes, n_cells)))'
-α = mean(group(chain_ppcaARD, :alpha))[:, 2]
-plot(α; label="alpha")
+W = permutedims(reshape(mean(group(chain_ppcaARD, :w))[:, 2], (n_genes, n_genes)))
+Z = permutedims(reshape(mean(group(chain_ppcaARD, :z))[:, 2], (n_genes, n_cells)))'
+α = mean(group(chain_ppcaARD, :α))[:, 2]
```

Comment:
Again it should be possible to simplify this.

Reply:
Yes, they are copied from the original version. 
A quick question: not quite sure what the `permutedims` is doing - IMO, ```W = reshape(mean(group(chain_ppcaARD, :W))[:, 2], (n_genes, n_genes))``` is enough; however, it will cause error later in this line ```df_pro[!, :type] = repeat([1, 2]; inner=n_cells ÷ 2)```, which means the dimensions are messed up without `permutedims`.
Any thoughts?